libs/clients/documents-v2/src/lib/documentsClientV2.service.ts

```diff
@@ -77,7 +84,25 @@ export class DocumentsClientV2Service {
       authenticationType: 'HIGH',
     })
 
-    return mapToDocument(document)
+    const mappedDocument = mapToDocument(document)
+
+    if (!mappedDocument) {
+      this.logger.warn('No document content available for findDocumentById', {
+        category: LOG_CATEGORY,
+        documentId,
+        documentProvider: document?.senderName ?? 'No provider available',
+      })
+      return null
+    }
+
+    if (!mappedDocument?.senderNationalId || !mappedDocument?.date) {
+      this.logger.debug('Document display data missing', {
+        category: LOG_CATEGORY,
+        document: document,
+      })
```

Comment:
Seems more important than just debug, doesn't this indicate an issue with the document?
Also, is it safe to log the entire document?